backend/worker/autoresolver.go

```diff
@@ -0,0 +1,86 @@
+package worker
+
+import (
+	"context"
+	"time"
+
+	"github.com/highlight-run/highlight/backend/model"
+	privateModel "github.com/highlight-run/highlight/backend/private-graph/graph/model"
+	"github.com/highlight-run/highlight/backend/store"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+type AutoResolver struct {
+	store *store.Store
+	db    *gorm.DB
+}
+
+func NewAutoResolver(store *store.Store, db *gorm.DB) *AutoResolver {
+	return &AutoResolver{
+		store: store,
+		db:    db,
+	}
+}
+
+func (autoResolver *AutoResolver) AutoResolveStaleErrors(ctx context.Context) {
+	projectFilterSettings := autoResolver.store.FindProjectsWithAutoResolveSetting(ctx)
+
+	for _, projectFilterSettings := range projectFilterSettings {
+		log.WithContext(ctx).WithFields(
+			log.Fields{"project_id": projectFilterSettings.ProjectID}).Info("Finding stale errors for project")
+		interval := projectFilterSettings.AutoResolveStaleErrorsDayInterval
+
+		project, err := autoResolver.store.GetProject(ctx, projectFilterSettings.ProjectID)
+		if err != nil {
+			log.WithContext(ctx).WithFields(log.Fields{"project_id": projectFilterSettings.ProjectID}).Error(err)
+			continue
+		}
+
+		err = autoResolver.resolveStaleErrorsForProjectInBatches(ctx, project, interval)
+
+		if err != nil {
+			log.WithContext(ctx).WithFields(log.Fields{"project_id": projectFilterSettings.ProjectID}).Error(err)
+			continue
+		}
+	}
+}
+
+func (autoResolver *AutoResolver) resolveStaleErrorsForProjectInBatches(ctx context.Context, project model.Project, interval int) error {
+	var errorGroups []model.ErrorGroup
+
+	db := autoResolver.db
+
+	subQuery := db.
+		Model(model.ErrorObject{}).
+		Select("error_group_id").
+		Where("created_at >= ?", time.Now().AddDate(0, 0, -interval))
+
+	result := db.Debug().Where(model.ErrorGroup{
+		State:     privateModel.ErrorStateOpen,
+		ProjectID: project.ID,
+	}).
+		Where("id NOT IN (?)", subQuery).
```

Comment:
i'm concerned this inverse subquery approach may break for projects with tons of new errors, since the subquery result is unbounded.

why do we need to use a subquery here? would something like this work:
```sql
select *
from error_groups eg
         inner join error_objects eo on eo.error_group_id = eg.id
where state = 'OPEN'
  and eg.project_id = 1
  and eg.created_at < NOW() - interval '30 day'
```


this will utilize some existing indexes and we can easily optimize further with a `state` and `created_at` index
<img width="1156" alt="Screenshot 2023-05-22 at 6 11 20 PM" src="https://github.com/highlight/highlight/assets/1351531/2fbfe4ac-6d47-4fd9-ae03-0e1b6dbdc46c">

Reply:
Adjusted in 87b02fa9f62a63b0cc1c4d7351dc80769769f0ab. Please check out the new query when you get a chance. Thanks!

Reply:
thanks! i believe the new query should use an `inner join` since we are filtering for `error_objects.created_at`. other than that, looks good.